packages/block-library/src/verse/style.scss

```diff
@@ -2,4 +2,8 @@ pre.wp-block-verse {
 	font-family: inherit;
 	overflow: auto;
 	white-space: pre-wrap;
+
+	// Prevent text overflow off-screen when the previous block has float.
+	min-width: 1em;
```

Comment:
This is a trick to prevent the text from being cut off. Without this property, characters will be cut off when the width of the verse text is less than one character.

![outside](https://user-images.githubusercontent.com/54422211/197332825-9a47a9be-9498-48e6-b995-b70cf761fc3f.png)